workshops/epsm-2022-ai-workshop.ipynb

```diff
@@ -0,0 +1,1758 @@
+    "    \"Lacrimal-Lt\": [TG263.Glnd_Lacrimal_L],\n",
+    "    \"Lacrimal-Rt\": [TG263.Glnd_Lacrimal_R],\n",
+    "    \"Lens-Lt\": [TG263.Lens_L],\n",
+    "    \"Lens-Rt\": [TG263.Lens_R],\n",
+    "    \"Optic-Nerve-Lt\": [TG263.OpticNrv_L],\n",
+    "    \"Optic-Nerve-Rt\": [TG263.OpticNrv_R],\n",
+    "    # If there was an \"Eyes\" structure, would do the following:\n",
+    "    # \"Eyes\": [TG263.Eye_L, TG263.Eye_R]\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "74a73803-96f5-48af-a3e1-4287407a5bfb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "structure_names = list(align_map.keys())\n",
+    "structure_names"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2104fc56-49b4-45b7-ab48-034eb1099bca",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dicom_contours_by_structure = rai.dicom_to_contours_by_structure(\n",
+    "    ds=structure_ds, image_uids=image_uids, structure_names=structure_names\n",
+    ")\n",
+    "dicom_contours_by_structure.keys()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f79231a5-b6bb-41d8-953e-c19d2a3eb9da",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "aligned_predicted_contours_by_structure = rai.merge_contours_by_structure(\n",
+    "    predicted_contours_by_structure, align_map\n",
+    ")\n",
+    "aligned_predicted_contours_by_structure.keys()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "288524c2-a52d-44cb-8f95-24dc40a0e35d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dice = {}\n",
+    "for name in align_map:\n",
+    "    dice[name] = rai.dice_from_contours_by_slice(\n",
+    "        dicom_contours_by_structure[name],\n",
+    "        aligned_predicted_contours_by_structure[name],\n",
+    "    )\n",
+    "\n",
+    "dice"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bc9f846a-8a3f-4297-b828-f9f5c5f0fb87",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "combined_contours_by_structure = {\n",
+    "    **predicted_contours_by_structure,\n",
+    "    **dicom_contours_by_structure,\n",
+    "}\n",
+    "\n",
+    "rai.plot_contours_by_structure(\n",
+    "    x_grid, y_grid, image_stack, combined_contours_by_structure, align_map\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8852c6a7-ace4-4e39-933c-ca168156e6f1",
+   "metadata": {},
+   "source": [
+    "## Building our own UNet"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6020af38-2575-4d06-b5b8-e8edddbf20d3",
+   "metadata": {
+    "id": "6020af38-2575-4d06-b5b8-e8edddbf20d3"
+   },
+   "source": [
+    "### Library imports\n",
+    "\n",
+    "Here are a set of library imports, from both the standard library and some libraries downloadable from PyPI. These are imported within namespaces so as not to variable and function name conflicts."
```

Comment:
```suggestion
    "Here are a set of library imports, from both the standard library and some libraries downloadable from PyPI. These are imported within namespaces to avoid variable and function name conflicts."
```